bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -1490,4 +1493,38 @@ private String describe(PullRequest pr) {
         return pr.repository().name() + "#" + pr.id();
     }
 
+    private boolean approvalNeeded() {
+        return approval != null && approval.needsApproval(pr.targetRef());
+    }
+
+    private void updateApprovalNeededComment(Map<String, Boolean> additionalProgresses) {
+        var existing = findComment(APPROVAL_NEEDED_MARKER);
+        var existingUnapproved = false;
+        for (var entry : additionalProgresses.entrySet()) {
+            if (entry.getKey().endsWith("needs maintainer approval") && !entry.getValue()) {
+                existingUnapproved = true;
+                break;
+            }
+        }
+        StringBuilder messageBuilder = new StringBuilder();
+        if (existingUnapproved) {
+            messageBuilder.append("⚠️  @").append(pr.author().username())
+                    .append(" There are still some issues that have not received maintainer approval.")
+                    .append("\n")
+                    .append("Please follow the instruction here to get maintainer approval: ")
+                    .append("[Requesting push approval for fixes](https://openjdk.org/projects/jdk-updates/approval.html)");
+        } else {
+            messageBuilder.append("@").append(pr.author().username()).append(" All the issues have already got the maintainer approval!");
```

Comment:
When looking at this, I think I'm changing my mind a bit around this. I think we should just post the "approval needed" comment once and not edit it with progress. It's enough to have progress in the pr body.